src/main/java/uk/gov/cslearning/record/api/InviteController.java

```diff
@@ -26,10 +28,24 @@ public InviteController(InviteService inviteService){
         return new ResponseEntity<>(result, HttpStatus.OK);
     }
 
-    @PostMapping("/{eventId}/invitee")
-    public ResponseEntity<Void> addInvitee(@PathVariable("eventId") String eventUid, @RequestBody InviteDto inviteDto, UriComponentsBuilder builder){
-        inviteService.save(inviteDto);
+    @GetMapping("/{eventId}/invitee/{inviteId}")
+    public ResponseEntity<InviteDto> getInvitee(@PathVariable("eventId") String eventUid, @PathVariable("inviteId") int inviteId){
+        return inviteService.findInvite(inviteId)
+                .map(i -> new ResponseEntity<>(i, HttpStatus.OK))
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    }
 
-        return ResponseEntity.created(builder.path("/event/{eventId}/invitee").build(eventUid)).build();
+    @PostMapping("/{eventId}/invitee")
+    public ResponseEntity<Object> addInvitee(@PathVariable("eventId") String eventUid, @Valid @RequestBody InviteDto inviteDto, BindingResult bindingResult, UriComponentsBuilder builder){
```

Comment:
This isn't quite what I had in mind. I was thinking this could be written as ...
```
    @PostMapping("/{eventId}/invitee")
    public ResponseEntity<Object> addInvitee(@PathVariable("eventId") String eventUid, @Valid @RequestBody InviteDto inviteDto, UriComponentsBuilder builder){
        return inviteService.save(inviteDto)
                .map(i -> ResponseEntity.created(builder.path("/event/{eventId}/invitee").build(eventUid)).build())
                .orElseThrow(() -> new IllegalStateException("Unable to save invite"));
    }
```
The ApiExceptionHandler already deals with validation errors [here](https://github.com/Civil-Service-Human-Resources/lpg-learner-record/blob/46643af1327b1aa37bd3fdef03253e4f8fb92134/src/main/java/uk/gov/cslearning/record/api/ApiExceptionHandler.java#L32l)

This will return a 400 with a body like...
```
{
    "timestamp": "2018-01-01T00:00:00",
    "errors": [
        "learnerEmail": "The learner must be registered"
    ],
    "status": 400,
    "message": "Bad Request"
}
```
You could then handle the error in lpg management using the message in the response body. This would be simpler to implement and keep the error handling in learner-record consistent.

Your implementation here introduces a need to check what error is returned every time an InviteDto is validated in order to return the correct response status. This is difficult to maintain (and test) and introduces a lot of code branches not just here but in every other method that validates an InviteDto.

If you don't agree with the approach above it would still be better to move the logic that determines whether to return a 409 response to the ApiExceptionHandler to save the need to duplicate it. The other check isn't needed.